source/parsing/Parser.cpp

```diff
@@ -82,7 +82,23 @@ AnsiPortListSyntax& Parser::parseAnsiPortList(Token openParen) {
                                                     TokenKind::Comma, closeParen,
                                                     RequireItems::False, diag::ExpectedAnsiPort,
                                                     [this] { return &parseAnsiPort(); });
-    return factory.ansiPortList(openParen, buffer.copy(alloc), closeParen);
+    AnsiPortListSyntax* portList = &factory.ansiPortList(openParen, buffer.copy(alloc), closeParen);
+
+    // put interface ports in metadata
+
+    for (auto port : portList->ports) {
+        if (port->kind != SyntaxKind::ImplicitAnsiPort)
+            continue;
+        ImplicitAnsiPortSyntax* ansiport = &port->as<ImplicitAnsiPortSyntax>();
+
+        if (ansiport->header->kind != SyntaxKind::InterfacePortHeader)
+            continue;
+        InterfacePortHeaderSyntax* goodheader = &ansiport->header->as<InterfacePortHeaderSyntax>();
+
+        meta.interfacePorts.push_back(goodheader);
+    }
+
+    return *portList;
 }
```

Comment:
This seems like a lot for the parser, should I maybe put this logic in the driver and use the nodemap?

Reply:
Why not just add to the interfacePorts list from the parsePortHeader function in the branch that creates InterfacePortHeaders directly so that you don't need this second pass through the list of ports?